### PR TITLE
feat(python): add  parameter to DeltaTable.to_pyarrow_dataset()

### PR DESCRIPTION
**Pull request by adriangb**
Otherwise there is no way to union this with another dataset.

### CODE REVIEW

**Comment by ion-elgreco** [comment]
@adriangb you might just better passthrough `as_large_types` to the `.to_pyarrow()`

**Comment by adriangb** [comment]
They both seem useful right? It seems like the `as_large_types` just blindly makes all types large. I could have a mix of small and large types of which I know the schema, so passing in the schema explicitly (given that it's simple to do so) seems worth having as an option.

**Comment by ion-elgreco** [comment]
@adriangb that's true. If you can fix the tests then we can merge

**Comment by adriangb** [comment]
it looks like the test just fails on older pyarrow versions and only for the map type. How about I split it in two and skip the failing one on pyarrow < 10?

**Comment by ion-elgreco** [comment]
@adriangb can you fix the tests? Then we can merge it :)

File: python/deltalake/table.py [hunk 1]
```diff
@@ -1022,6 +1022,8 @@ def to_pyarrow_dataset(
         partitions: Optional[List[Tuple[str, str, Any]]] = None,
         filesystem: Optional[Union[str, pa_fs.FileSystem]] = None,
         parquet_read_options: Optional[ParquetReadOptions] = None,
+        schema: Optional[pyarrow.Schema] = None,
+        as_large_types: bool = False,
```
Comment (ion-elgreco) [inline python/deltalake/table.py:1026]: Doc description is missing for this param. I would also mention if the schema is passed that takes precedence over as_large_types

**Review by ion-elgreco — CHANGES_REQUESTED** [review]

File: python/deltalake/table.py (hunk 1 above)
Comment (adriangb) [reply python/deltalake/table.py:1026]: done!

**Review by ion-elgreco — APPROVED** [review]

**Comment by ion-elgreco** [comment]
Thankss @adriangb

**Pull request merged**